server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Email.scala

```diff
@@ -84,14 +86,28 @@ object Email {
         bodyStructure = bodyStructure,
         textBody = bodyStructure.textBody,
         htmlBody = bodyStructure.htmlBody,
-        attachments = bodyStructure.attachments)
+        attachments = bodyStructure.attachments,
+        asEmailHeaders(mime4JMessage.getHeader)
+      )
     }
   }
+
+  private def asEmailHeaders(header: Header): List[EmailHeader] = {
+    header.iterator()
+      .asScala
+      .map(header => EmailHeader(
+        EmailHeaderName(header.getName),
+        EmailHeaderValue(new String(header.getRaw.toByteArray)
```

Comment:
Do we need to know the header encoding ?

Reply:
Always US-ASCII ?

Reply:
But we need to specify explicitly the charset!